### PR TITLE
Check updates on Application onStart event, tests

### PR DESCRIPTION
**Pull request by Jamminroot**
Please have a look at our [guidelines for contributions](https://github.com/microsoft/appcenter-sdk-android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

This PR covers `ApplicationLifecycleListener` class, fixes coverage of `AbstractAppCenterService` and adds call to check for updates on Application lifecycle onStart event.

## Related PRs or issues

[AB#78294](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/78294)

### CODE REVIEW

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeTest.java [hunk 1]
```diff
@@ -93,6 +93,11 @@ public void checkFactories() {
         assertTrue(factories.isEmpty());
     }
 
+    @Test
+    public void onApplicationStartTest() {
```
Comment (MatkovIvan) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeTest.java]: It seems not related (and empty)

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/AbstractAppCenterServiceTest.java [hunk 2]
```diff
@@ -92,6 +92,14 @@ public Object answer(InvocationOnMock invocation) {
         SharedPreferencesManager.putBoolean(eq(SERVICE_ENABLED_KEY), anyBoolean());
     }
 
+    @Test
+    public void onApplicationStarted() {
+        mService.onApplicationStarted();
+    }
```
Comment (MatkovIvan) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/AbstractAppCenterServiceTest.java]: What does it test?

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/AbstractAppCenterServiceTest.java [hunk 3]
```diff
@@ -92,6 +92,14 @@ public Object answer(InvocationOnMock invocation) {
...
+
+    @Test
+    public void onApplicationStopped() { mService.onApplicationStopped(); }
```
Comment (MatkovIvan) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/AbstractAppCenterServiceTest.java]: Format

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+@PrepareForTest({Handler.class})
+public class ApplicationLifecycleListenerTest {
+
+    private Handler mHandlerMock;
+    private ApplicationLifecycleListener mApplicationLifecycleListenerMock;
+    private Activity mActivityMock;
+
+    @Before
+    public void setUp() {
+        mHandlerMock = mock(Handler.class);
```
Comment (MatkovIvan) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java]: Use @Mock instead

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+        mApplicationLifecycleListenerMock = new ApplicationLifecycleListener(mHandlerMock);
+        mActivityMock = mock(Activity.class);
```
Comment (MatkovIvan) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java]: Same

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+            }
+        }).when(mHandlerMock).removeCallbacks(any(Runnable.class));
+    }
+
+    @After
+    public void tearDown() {
+        mHandlerMock = null;
+        mApplicationLifecycleListenerMock = null;
+        mActivityMock = null;
+    }
```
Comment (MatkovIvan) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java]: Seems useless

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/AbstractAppCenterServiceTest.java (hunk 2 above)
Comment (Jamminroot) [reply sdk/appcenter/src/test/java/com/microsoft/appcenter/AbstractAppCenterServiceTest.java]: Coverage fix

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeTest.java (hunk 1 above)
Comment (Jamminroot) [reply sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeTest.java]: Coverage fix

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+        mApplicationLifecycleListenerMock.onActivityResumed(mActivityMock);
+        verify(mHandlerMock, never()).removeCallbacks(any(Runnable.class));
+
+        /* Call onActivityResumed again. */
+        mApplicationLifecycleListenerMock.onActivityResumed(mActivityMock);
+        verify(mHandlerMock, never()).removeCallbacks(any(Runnable.class));
+
+        /* Pause.*/
+        mApplicationLifecycleListenerMock.onActivityPaused(mActivityMock);
+        verify(mHandlerMock, never()).postDelayed(any(Runnable.class), anyLong());
```
Comment (annakocheshkova) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java]: Let's be more specific, we have a const that can be used instead of `anyLong`

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java (hunk 6 above)
Comment (annakocheshkova) [reply sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java]: Doesn't it have `stopMocking` or something like this?

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/AbstractAppCenterServiceTest.java (hunk 2 above)
Comment (MatkovIvan) [reply sdk/appcenter/src/test/java/com/microsoft/appcenter/AbstractAppCenterServiceTest.java]: Did you verify subscription on these events? Please cover this part properly

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java (hunk 7 above)
Comment (MatkovIvan) [reply sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java]: @annakocheshkova disagree. it's about "never" check

Comment (annakocheshkova) [reply sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java]: Well it exists in other places, too

**Comment by MatkovIvan** [comment]
Tests for core changes must be a part of #1385. Please open a separate PR with distribute module changes
